SDWebImage/NSData+ImageContentType.m

```diff
@@ -33,16 +34,26 @@ + (SDImageFormat)sd_imageFormatForImageData:(nullable NSData *)data {
         case 0x49:
         case 0x4D:
             return SDImageFormatTIFF;
-        case 0x52:
-            // R as RIFF for WEBP
-            if (data.length < 12) {
-                return SDImageFormatUndefined;
+        case 0x52: {
+            if (data.length >= 12) {
+                //RIFF....WEBP
+                NSString *testString = [[NSString alloc] initWithData:[data subdataWithRange:NSMakeRange(0, 12)] encoding:NSASCIIStringEncoding];
+                if ([testString hasPrefix:@"RIFF"] && [testString hasSuffix:@"WEBP"]) {
+                    return SDImageFormatWebP;
+                }
             }
-            
-            NSString *testString = [[NSString alloc] initWithData:[data subdataWithRange:NSMakeRange(0, 12)] encoding:NSASCIIStringEncoding];
-            if ([testString hasPrefix:@"RIFF"] && [testString hasSuffix:@"WEBP"]) {
-                return SDImageFormatWebP;
+            break;
+        }
+        case 0x00: {
+            if (data.length >= 12) {
+                //....ftypheic
+                NSString *testString = [[NSString alloc] initWithData:[data subdataWithRange:NSMakeRange(4, 8)] encoding:NSASCIIStringEncoding];
+                if ([testString hasPrefix:@"ftyp"] && [testString hasSuffix:@"heic"]) {
```

Comment:
Well. Maybe this could change to [isEqualToString: @"ftypheic"]. A little misunderstanding

Reply:
Yeap, `[isEqualToString: @"ftypheic"]` is better.

Reply:
But not a blocker for merging this PR.